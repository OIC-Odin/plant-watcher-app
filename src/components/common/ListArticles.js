import React from 'react';
import { ListItem, Error, Spinner } from './';
import { FlatList } from 'react-native';

const renderRow = ({ plantName, articleContents, image }) => <ListItem plantName={plantName} articleContents={articleContents} image={image} />;

export const ListArticles = ({ articles, loading, error }) => {
  if (loading) return <Spinner size='large' />;
  if (error) return <Error message={error} />;
  return (
    <FlatList
      data={articles}
      renderItem={({ item }) => renderRow(item)}
    />
  );
};
