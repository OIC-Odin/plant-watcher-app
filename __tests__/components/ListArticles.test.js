import React from 'react';
import ListArticles from '../../src/components/ListArticles';
import renderer from 'react-test-renderer';

describe('ListArticles', () => {
  it('snapshot-記事が見つからない時', () => {
    const INITIAL_STATE = {
      articles: [],
      error: '記事が見つかりません',
      loading: false,
    };
    const tree = renderer.create(<ListArticles {...INITIAL_STATE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('snapshot-ロード中', () => {
    const INITIAL_STATE = {
      articles: [],
      error: '',
      loading: true
    };
    const tree = renderer.create(<ListArticles {...INITIAL_STATE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('snapshot-記事が表示されている時', () => {
    const INITIAL_STATE = {
      articles: [{
        plantName: 'ひまわり',
        articleContents: 'きいろ\nまるい',
        key: 'esyBvhxqwxP5UjaUeQyokRszulJ3',
        createdAt: 1514994952976,
        uid: 'brURQVswxEZf2z42nJESTwLaDfr2'
      }],
      error: '',
      loading: true
    };
    const tree = renderer.create(<ListArticles {...INITIAL_STATE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
