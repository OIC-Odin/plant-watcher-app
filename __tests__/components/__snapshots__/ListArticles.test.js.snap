// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`記事の表示 snapshot-正しく画面が呼び出される 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "articleContents": "this is test contents. 
     this contents is test data.",
        "articleId": "-KxRJa7r69I46Xey3MnO",
        "key": "esyBvhxqwxP5UjaUeQyokRszulJ3",
        "plantName": "テスト",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  onViewableItemsChanged={undefined}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderColor": "#e1e8ee",
              "borderWidth": 1,
              "margin": 15,
              "marginBottom": 0,
              "padding": 15,
              "shadowColor": "rgba(0,0,0, .2)",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 1,
            },
            Object {
              "padding": 0,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
              undefined,
            ]
          }
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "#43484d",
                      "fontSize": 17.5,
                      "fontWeight": "bold",
                      "marginBottom": 15,
                      "textAlign": "center",
                    },
                    Object {
                      "marginTop": 15,
                    },
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              テスト
            </Text>
          </View>
          <View
            style={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "height": 150,
                    "width": null,
                  },
                  undefined,
                ]
              }
            >
              <Image
                resizeMode="cover"
                source={1}
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Object {
                      "height": undefined,
                      "width": undefined,
                    },
                    undefined,
                  ]
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "alignSelf": "stretch",
                      "backgroundColor": "rgba(0, 0, 0, 0.2)",
                      "bottom": 0,
                      "flex": 1,
                      "justifyContent": "center",
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {},
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      Array [
                        Object {
                          "color": "white",
                          "fontSize": 22.5,
                          "fontWeight": "800",
                          "marginBottom": 8,
                        },
                        undefined,
                      ],
                    ]
                  }
                />
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {},
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      Array [
                        Object {
                          "color": "white",
                          "fontSize": 16.25,
                          "fontWeight": "400",
                          "marginBottom": 8,
                        },
                        undefined,
                      ],
                    ]
                  }
                />
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "padding": 10,
                  },
                  undefined,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Object {
                    "paddingBottom": 8,
                  }
                }
              >
                this is test contents. 
                 this contents is test data.
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "marginLeft": 15,
                      "marginRight": 15,
                    },
                    undefined,
                    undefined,
                    undefined,
                  ]
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hasTVPreferredFocus={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                      },
                      undefined,
                    ]
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#9E9E9E",
                          "flexDirection": "row",
                          "justifyContent": "center",
                          "padding": 19,
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        Object {
                          "backgroundColor": "#228b22",
                        },
                        undefined,
                        Object {
                          "padding": 12,
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                      ]
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={false}
                      disabled={false}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "white",
                            "fontSize": 18,
                          },
                          Array [
                            Object {
                              "marginRight": 10,
                            },
                            undefined,
                          ],
                          Object {
                            "fontFamily": "simple-line-icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                        ]
                      }
                      type="simple-line-icon"
                    >
                      
                    </Text>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      disabled={false}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {},
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          Array [
                            Object {
                              "color": "white",
                              "fontSize": 20,
                            },
                            undefined,
                            Object {
                              "fontSize": 17.5,
                            },
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                          ],
                        ]
                      }
                    >
                      読む
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
